utils/tx.ts

```diff
@@ -425,20 +426,17 @@ export const createPool = async (
 // for alignment purposes lets keep it backward comaptible
 // so every pool will have same weight
 export const promotePool = async (
-  sudoAccount: KeyringPair,
+  sudoAccount: User,
   liqAssetId: BN,
   weight: number = 100,
 ) => {
   testLog.getLog().info(`Promoting pool :${liqAssetId}`);
   const mangata = await getMangataInstance();
   const api = await mangata.api();
-  return await signTx(
-    api,
-    api.tx.sudo.sudo(
-      api.tx.proofOfStake.updatePoolPromotion(liqAssetId, weight),
-    ),
-    sudoAccount,
-    { nonce: await getCurrentNonce(sudoAccount.address) },
+  return await signTxMetamask(
+    api.tx.proofOfStake.updatePoolPromotion(liqAssetId, weight),
+    sudoAccount.ethAddress.toString(),
+    sudoAccount.name.toString(),
```

Comment:
I think that this can not work. 
+ updatePoolPromotion requires api.tx.sudo.sudo( updatePoolPromotion ) 
+ sudo ALWAYS require the sudo nonce. 

Plz, double check if tests are working and using this function, perhaps dead code? 

Additionally, you can still use the regular signTx with eth addresses, don't you?

Reply:
Issues was fixed